SplunkAppForWazuh/appserver/static/js/controllers/overview/aws/awsCtrl.js

```diff
@@ -2,74 +2,156 @@ define([
   '../../module',
   '../../../services/visualizations/chart/pie-chart',
   '../../../services/visualizations/chart/area-chart',
+  '../../../services/visualizations/chart/column-chart',
   '../../../services/visualizations/table/table',
   '../../../services/visualizations/map/map',
-  '../../../services/visualizations/inputs/time-picker'
-], function(app, PieChart, AreaChart, Table, Map, TimePicker) {
+  '../../../services/visualizations/inputs/time-picker',
+  '../../../services/visualizations/inputs/dropdown-input'
+], function (app, PieChart, AreaChart, ColumnChart, Table, Map, TimePicker, Dropdown) {
   'use strict'
 
   class AWS {
-    constructor($urlTokenModel, $scope, $currentDataService, $state) {
+    constructor($urlTokenModel, $rootScope, $scope, $currentDataService, $state, awsMetrics) {
+      this.rootScope = $rootScope
       this.scope = $scope
       this.state = $state
+      this.awsMetrics = awsMetrics
       this.currentDataService = $currentDataService
       this.currentDataService.addFilter(`{"rule.groups":"amazon", "implicit":true}`)
       this.getFilters = this.currentDataService.getSerializedFilters
       this.filters = this.getFilters()
+      this.amazonFilters = this.buildAmazonFilter()
+      this.implicitFilters = this.serializedImplicitFilters(this.currentDataService.getFilters().filter(item => item['implicit']))
       this.submittedTokenModel = $urlTokenModel.getSubmittedTokenModel()
       this.timePicker = new TimePicker(
         '#timePicker',
         $urlTokenModel.handleValueChange
       )
+
+      if (this.awsMetrics == false) {
+        this.scope.awsMetrics = {
+          enabled: "N/D",
+          scanInterval: "N/D",
+          runOnStart: "N/D",
+          skipOnError: "N/D",
+          buckets: "N/D",
+          accountInUse: "N/D",
+          regionsInUse: "N/D"
+        }
+      } else {
+        this.awsMetrics = this.awsMetrics[0]['aws-s3']
+        this.scope.awsMetrics = {
+          enabled: "",
+          scanInterval: this.awsMetrics.interval,
+          runOnStart: this.awsMetrics.run_on_start,
+          skipOnError: this.awsMetrics.skip_on_error,
+          buckets: "",

```

Comment:
`buckets` must be an array because next iteration takes a single bucket for additional features.

Reply:
Same for accounts, regions, etc...